include/openssl/macros.h

```diff
@@ -19,6 +19,9 @@
 # define OPENSSL_MSTR_HELPER(x) #x
 # define OPENSSL_MSTR(x) OPENSSL_MSTR_HELPER(x)
 
+/* Check if pre, which must be a string literal, is a prefix of str */
+# define HAS_PREFIX(str, pre) (strncmp(str, pre "", sizeof(pre) - 1) == 0)
```

Comment:
I don't see why we make this a public macro, but most of all, if it should be a public macro, it must be named with `OSSL_` or `OPENSSL_`

Reply:
It shouldn't be public.

Reply:
I also think a public macro would be better/sufficient. 
Where to place it to make it non-public but still accessible from libs, apps, and tests?

Reply:
> Where to place it to make it non-public but still accessible from libs, apps, and tests?

Uhmmm, in `include/internal`, and then you `#include "internal/whatever.h"`

Reply:
Looks like there is no generally included non-public header file.
So I moved the `#define` to `include/internal/cryptlib.h`
and added its inclusion as far as needed to a couple of rather generally included headers, 
namely `apps.h`, `testutil.h`, and `ssl_local.h`.

Reply:
Or how about adding `include/internal/general.h`
and moving there all decls that are not specific to crypto?

Reply:
Again my question:

How about introducing `include/internal/general.h`, moving there all decls form `cryptlib.h` that are not crypto specific,
and including `internal/general.h` from `cryptlib.h`, `ssl_local.h.` `apps.h,` and `testutil.h?`

Reply:
internal/common.h?

Reply:
Moved the more general parts of `internal/cryptlib.h` to a new `internal/common.h`.